pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/BaseBrokerRequestHandler.java

```diff
@@ -487,6 +491,11 @@ private BrokerResponseNative handleSQLRequest(long requestId, String query, Json
       _brokerMetrics.addMeteredTableValue(rawTableName, BrokerMeter.BROKER_RESPONSES_WITH_NUM_GROUPS_LIMIT_REACHED, 1);
     }
 
+    if (0 != numUnavailableSegments) {
```

Comment:
variable `numUnavailableSegments` seems redundant, use 
* `requestStatistics.setNumUnavailableSegments(unavailableSegments.size())` at L445
* `if (! unavailableSegments.isEmpty())` at L494
Same for other places

Reply:
Removed `numUnavailableSegments`